plugins/diff/erofs/differ_linux.go

```diff
@@ -54,10 +62,152 @@ func NewErofsDiffer(store content.Store, mkfsExtraOpts []string) differ {
 	}
 }
 
+func writeDiff(ctx context.Context, w io.Writer, lower []mount.Mount, upperRoot string) error {
+	var opts []archive.ChangeWriterOpt
+
+	return mount.WithTempMount(ctx, lower, func(lowerRoot string) error {
+		cw := archive.NewChangeWriter(w, upperRoot, opts...)
+		err := fs.DiffDirChanges(ctx, lowerRoot, upperRoot, fs.DiffSourceOverlayFS, cw.HandleChange)
+		if err != nil {
+			return fmt.Errorf("failed to create diff tar stream: %w", err)
+		}
+		return cw.Close()
+	})
+}
+
 // Compare creates a diff between the given mounts and uploads the result
 // to the content store.
 func (s erofsDiff) Compare(ctx context.Context, lower, upper []mount.Mount, opts ...diff.Opt) (d ocispec.Descriptor, err error) {
-	return emptyDesc, fmt.Errorf("erofsDiff does not implement Compare method: %w", errdefs.ErrNotImplemented)
+	layer, err := erofsutils.MountsToLayer(upper)
+	if err != nil {
+		return emptyDesc, fmt.Errorf("unsupported layer for erofsDiff Compare method: %w", errdefs.ErrNotImplemented)
```

Comment:
`MountsToLayer` already returns a wrapped `ErrNotImplemented` with a more specific message

Reply:
ok, thanks derek. let me fix it now since it doesn't merge for now.

Reply:
fixed.